inc/functions-icons.php

```diff
@@ -0,0 +1,206 @@
+ * @return string  $item_output The menu item output with social icon.
+ */
+function twentyseventeen_nav_menu_social_icons( $item_output, $item, $depth, $args ) {
+
+	// Get supported social icons.
+	$social_icons = twentyseventeen_social_links_icons();
+
+	// Change SVG icon inside social links menu if there is supported URL.
+	if ( 'social' == $args->theme_location ) {
+		foreach ( $social_icons as $attr => $value ) {
+			if ( false !== strpos( $item_output, $attr ) ) {
+				$item_output = str_replace( $args->link_after, '</span>' . twentyseventeen_get_svg( array( 'icon' => esc_attr( $value ) ) ), $item_output );
+			}
+		}
+	}
+
+	return $item_output;
+
+}
+add_filter( 'walker_nav_menu_start_el', 'twentyseventeen_nav_menu_social_icons', 10, 4 );
+
+/**
+ * Add dropdown icon if menu item has children.
+ *
+ * @param string  $title The menu item's title.
+ * @param object  $item  The current menu item.
+ * @param array   $args  An array of wp_nav_menu() arguments.
+ * @param int     $depth Depth of menu item. Used for padding.
+ * @return string $title The menu item's title with dropdown icon.
+ */
+function twentyseventeen_dropdown_icon_to_menu_link( $title, $item, $args, $depth ) {
+
+	if ( 'top' == $args->theme_location ) {
+
+		foreach ( $item->classes as $value ) {
+			if ( $value == 'menu-item-has-children' || $value == 'page_item_has_children' ) {
+				$title =  $title . twentyseventeen_get_svg( array( 'icon' => 'expand' ) );
+			}
+		}
+
+	}
+
+	return $title;
+
+}
+add_filter( 'nav_menu_item_title', 'twentyseventeen_dropdown_icon_to_menu_link', 10, 4 );
+
+/**
+ * Returns an array of supported social links (URL and icon name).
+ *
+ * @return array $social_links_icons
+ */
+function twentyseventeen_social_links_icons() {
+
+	// Supported social links icons.
+	$social_links_icons = array(
+		'codepen.io'      => 'codepen',
+		'digg.com'        => 'digg',
+		'dribbble.com'    => 'dribbble',
+		'dropbox.com'     => 'dropbox',
+		'facebook.com'    => 'facebook',
+		'flickr.com'      => 'flickr',
+		'foursquare.com'  => 'foursquare',
+		'plus.google.com' => 'google-plus',
+		'github.com'      => 'github',
+		'instagram.com'   => 'instagram',
+		'linkedin.com'    => 'linkedin',
+		'mailto:'         => 'envelope-o',
+		'path.com'        => 'path',
+		'pinterest.com'   => 'pinterest-p',
+		'getpocket.com'   => 'get-pocket',
+		'polldaddy.com'   => 'polldaddy',
+		'reddit.com'      => 'reddit-alien',
+		'skype.com'       => 'skype',
+		'skype:'          => 'skype',
+		'soundcloud.com'  => 'soundcloud',
+		'spotify.com'     => 'spotify',
+		'stumbleupon.com' => 'stumbleupon',
+		'tumblr.com'      => 'tumblr',
+		'twitch.tv'       => 'twitch',
+		'twitter.com'     => 'twitter',
+		'vimeo.com'       => 'vimeo',
+		'wordpress.org'   => 'wordpress',
+		'wordpress.com'   => 'wordpress',
+		'youtube.com'     => 'youtube',
+	);
+
+	return apply_filters( 'twentyseventeen_social_links_icons', $social_links_icons );
```

Comment:
Considering we are talking about switching to a child-theme-able setup, I think my comment here doesn't apply - users will be able to add their own SVG file, so they will want to be able to filter this. 👍